aea/configurations/loader.py

```diff
@@ -199,28 +200,27 @@ def load(self, file_pointer: TextIO) -> T:
         :param file_pointer: the file pointer to the configuration file
         :return: the configuration object.
         """
-        if self.configuration_class.package_type == PackageType.AGENT:
-            return cast(T, self._load_agent_config(file_pointer))
 
-        if self.configuration_class.package_type == PackageType.SERVICE:
-            return cast(T, self._load_service_config(file_pointer))
+        loader: Callable = {
+            PackageType.AGENT: self._load_agent_config,
+            PackageType.SERVICE: self._load_service_config,
+        }.get(self.configuration_class.package_type, self._load_component_config)
 
-        return self._load_component_config(file_pointer)
```

Comment:
Please avoid updating code in test only PRs, open and issue to deal with this later and try to keep the PR strictly for test coverage

Reply:
Agreed!